test/e2e/l7networkpolicy_test.go

```diff
@@ -434,12 +441,35 @@ type L7LogEntry struct {
 	DestPort    int32            `json:"dest_port"`
 	Protocol    string           `json:"proto"`
 	AppProtocol string           `json:"app_proto,omitempty"`
+	Packet      string           `json:"packet"`
 	Http        *L7LogHttpEntry  `json:"http,omitempty"`
 	Alert       *L7LogAlertEntry `json:"alert,omitempty"`
 }
 
+// Compares if two L7LogEntry are equal. They must be exactly equal if Packet is
+// in the same base. Otherwise, if either of the Packet is in base64, they equal
+// if Packet in base64 contains essential message in other Packet.
 func (e *L7LogEntry) Equal(x *L7LogEntry) bool {
-	return reflect.DeepEqual(e, x)
+	packetLooseEqual := conversion.EqualitiesOrDie(
+		func(a, b L7LogEntry) bool {
```

Comment:
unless I am missing something, this is incorrect because it completely ignores the other fields in `L7LogEntry` and only looks at the `Packet` field (all other fields are ignored in the comparison)?

I don't know exactly how `conversion.EqualitiesOrDie` is supposed to be used, but in this case it may be easier / more readable to just write the comparison manually:
```
return e1.EventType == e2.EventType && ... && packetMatches && reflect.DeepEqual(e1.Http, e2.Http) && reflect.DeepEqual(e1.Alert, e2.Alert)
```